### PR TITLE
fix Volume migration across pools sometimes loses volume

### PR DESCRIPTION
**Pull request by div8cn**
## Description

When secondary storage usage is> 90%, VOLUME migration across primary storage will cause the migration to fail and lose VOLUME

Fixes: #3958

## Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

### CODE REVIEW

File: engine/storage/datamotion/src/main/java/org/apache/cloudstack/storage/motion/AncientDataMotionStrategy.java [hunk 1]
```diff
@@ -486,6 +486,9 @@ public void copyAsync(DataObject srcData, DataObject destData, Host destHost, As
             if (answer != null && !answer.getResult()) {
                 errMsg = answer.getDetails();
             }
+            if (answer == null) {
+                errMsg = "answer is null, set to error for CopyCommandResult";
```
Comment (DaanHoogland) [inline engine/storage/datamotion/src/main/java/org/apache/cloudstack/storage/motion/AncientDataMotionStrategy.java]: I don't understand this error message. Can you make it very clear, please?
for example "no answer from came back from copyCommand, considering it as failed."
preferably even with some data like src, dest, src.type and dest.type

File: engine/storage/datamotion/src/main/java/org/apache/cloudstack/storage/motion/AncientDataMotionStrategy.java [hunk 2]
```diff
@@ -348,8 +348,10 @@ protected Answer copyVolumeBetweenPools(DataObject srcData, DataObject destData)
             // directly to s3
             ImageStoreEntity imageStore = (ImageStoreEntity)dataStoreMgr.getImageStoreWithFreeCapacity(destScope.getScopeId());
             if (imageStore == null || !imageStore.getProtocol().equalsIgnoreCase("nfs") && !imageStore.getProtocol().equalsIgnoreCase("cifs")) {
-                s_logger.debug("can't find a nfs (or cifs) image store to satisfy the need for a staging store");
-                return null;
+                //s_logger.debug("can't find a nfs (or cifs) image store to satisfy the need for a staging store");
```
Comment (DaanHoogland) [inline engine/storage/datamotion/src/main/java/org/apache/cloudstack/storage/motion/AncientDataMotionStrategy.java:351]: please remove commented code

**Comment by DaanHoogland** [comment]
code looks good, let's get this tested and applied to 4.13 !

**Comment by DaanHoogland** [comment]
@div8cn can you rebase this commit on 4.13 so we can apply it there. We'll fwd merge it afterwards

**Comment by borisstoyanov** [comment]
@blueorangutan package

**Comment by DaanHoogland** [comment]
closing in favour of #4043

**Pull request closed without merging**